packages/swap/src/routes/v2/quote.ts

```diff
@@ -225,25 +230,50 @@ export const generateQuotes = async ({
     quoteType: 'REGULAR' as const,
   };
 
-  const [quoteResult, boostedQuoteResult, dcaQuoteResult, dcaBoostedQuoteResult] =
-    await Promise.allSettled([
-      getPoolQuote(quoteArgs),
-      estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
-      dcaQuoteParams &&
-        getPoolQuote({
-          ...quoteArgs,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-      dcaQuoteParams &&
-        estimatedBoostFeeBps &&
-        getPoolQuote({
-          ...quoteArgs,
-          boostFeeBps: estimatedBoostFeeBps,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-    ]);
+  const [quoteResult, boostedQuoteResult] = await Promise.allSettled([
+    getPoolQuote(quoteArgs),
+    estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
+  ]);
+  const ingressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'INGRESS')
+      : undefined;
+  const egressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'EGRESS')
+      : undefined;
+
+  const ingressFeeSurcharge = dcaQuoteParams
+    ? BigInt(
+        new BigNumber(ingressFee?.amount ?? 0)
+          .multipliedBy((dcaQuoteParams.numberOfChunks - 1) / dcaQuoteParams.numberOfChunks)
+          .toFixed(0),
+      )
+    : 0n;
+  const egressFeeSurcharge = dcaQuoteParams
```

Comment:
think we dont need to calculate this here but can do it inside the `adjustDcaQuote` (all data should be on the `dcaQuote` already)